src/resolvers/git.cr

```diff
@@ -214,20 +215,24 @@ module Shards
       ref = git_ref(version)
 
       Dir.mkdir_p(install_path)
-      run "git --work-tree=#{Process.quote(install_path)} checkout #{ref.to_git_ref} -- ."
+      run "git --work-tree=#{Process.quote(install_path)} checkout #{Process.quote(ref.to_git_ref)} -- ."
     end
 
     def commit_sha1_at(ref : GitRef)
-      capture("git log -n 1 --pretty=%H #{ref.to_git_ref}").strip
+      capture("git log -n 1 --pretty=%H #{Process.quote(ref.to_git_ref)}").strip
     end
 
     def local_path
       @local_path ||= begin
         uri = parse_uri(git_url)
 
-        path = uri.path.to_s[1..-1]
-        path = path.gsub('/', File::SEPARATOR) unless File::SEPARATOR == '/'
+        path = uri.path
         path += ".git" unless path.ends_with?(".git")
+        path = Path[path]
+        if (anchor = path.anchor)
+          path = Path[path.drive.to_s.rchop(":"), path.relative_to(anchor)]
```

Comment:
Comment done.

One thing I'm really sure of is that no, `join` should never consider putting the anchor in the middle of the path. It's doing it at the moment, though, just that nobody (except me) had given it a second thought before seeing this edge case. Respectfully, `join`'s behavior on Windows is more useful than POSIX because at least in almost all cases it will fail with a loud bang (unfortunately indirectly though).

The fact that this strange operation is explicitly chosen here doesn't mean anything for general usefulness of this. It should stay explicit.

Reply:
Yeah, but maybe a method to transform an anchored path into a safe-to-use relative path could be a useful tool.